lib/bdd/rspec/formatter.rb

```diff
@@ -29,45 +33,40 @@ def example_failed(failure)
 
     private
 
-      def read_steps(example, color2=nil)
-        last_step_title = ""
-        example.metadata[:step_messages].map do |hash|
-          msg   = hash[:msg]
-          color = hash[:color] || color2 || :light_black
-          if msg.is_a? Array
-            msg0 =  if msg[0] == last_step_title
-                      blank_step_title
-                    else
-                      text_with_color(msg[0], :white)
-                    end
-            last_step_title = msg[0]
+      def read_steps(example, text_color)
+        last_step_prefix = ""
+        prefix_length    = example.metadata[:bdd_prefix_max_length]
 
-            msg = [msg0, text_with_color(msg[1], color)].join(' ')
+        example.metadata[:bdd_step_messages].map do |msg|
+          if
+            msg.is_a? Array
+          then
+            prefix, text = msg
```

Comment:
no, msg will always be an Array, originally the code was longer and the prefix did not come with the msg.
Now, always array.